dummyserver/testcase.py

```diff
@@ -115,13 +115,16 @@ class HTTPDummyServerTestCase(unittest.TestCase):
     scheme = 'http'
     host = 'localhost'
     host_alt = '127.0.0.1'  # Some tests need two hosts
-    certs = DEFAULT_CERTS
+
+    @classmethod
```

Comment:
Could be mistaken because I've never tried it but properties aren't available to the class only the instance?  We need it for `_start_server()` which is also a `classmethod`.

Reply:
Just curious why the change was necessary, I assumed the old `certs` would be available to the class too.

Reply:
I used a function here because we're making a copy of `DEFAULT_CERTS` and changing the `ssl_version` key per-test for the TLS version tests.

Reply:
So you're saying before we'd have to do something like...

```python
tlsv1_certs = DEFAULT_CERTS.copy()
tlsv1_certs['ssl_version'] = ssl.PROTOCOL_TLSv1

class TestHTTPS_TLSv1(TestHTTPS_TLSVersion):
    tls_protocol_name = 'TLSv1'
    certs = tlsv1_certs
```

Instead of...

```python
class TestHTTPS_TLSv1(TestHTTPS_TLSVersion):
    tls_protocol_name = 'TLSv1'

     @classmethod
    def certs(cls):
        certs = DEFAULT_CERTS.copy()
        certs['ssl_version'] = ssl.PROTOCOL_TLSv1
        return certs
```

?

Not obvious to me what the advantage of the latter one is, but I don't have strong feelings here. :)